container/libcontainer/handler.go

```diff
@@ -60,9 +60,23 @@ func NewHandler(cgroupManager cgroups.Manager, rootFs string, pid int, includedM
 
 // Get cgroup and networking stats of the specified container
 func (h *Handler) GetStats() (*info.ContainerStats, error) {
-	cgroupStats, err := h.cgroupManager.GetStats()
-	if err != nil {
-		return nil, err
+	var cgroupStats *cgroups.Stats
+	readCgroupStats := true
+	if cgroups.IsCgroup2UnifiedMode() {
+		p, err := h.cgroupManager.GetUnifiedPath()
+		if err != nil {
+			return nil, err
+		}
+		if p == fs2.UnifiedMountpoint {
```

Comment:
what is this checking, and why do we not want to read cgroup stats anymore?

Reply:
it is checking if it is the root cgroup, if it can help I can change the variable name to make it more explicit.
With cgroup v2 there are no stats in the root cgroup, foe example:

```
$ ls /sys/fs/cgroup/memory.stat /sys/fs/cgroup/init.scope/memory.stat
ls: cannot access '/sys/fs/cgroup/memory.stat': No such file or directory
/sys/fs/cgroup/init.scope/memory.stat
```

Reply:
a comment would also be helpful

Reply:
thanks for the review.  Added a comment